src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEntryPointSymbol.cs

```diff
@@ -452,13 +421,18 @@ private sealed class ScriptEntryPoint : SynthesizedEntryPointSymbol
             //     var script = new Script();
             //     script.<Initialize>().GetAwaiter().GetResult();
             // }
-            internal override BoundBlock CreateBody()
+            internal override BoundBlock CreateBody(DiagnosticBag diagnostics)
             {
-                // CreateBody should only be called if no errors.
-                Debug.Assert((object)_getAwaiterMethod != null);
-                Debug.Assert((object)_getResultMethod != null);
-
                 var syntax = DummySyntax();
+                var compilation = _containingType.DeclaringCompilation;
+
+                // Creates a new top-level binder that just contains the global imports for the compilation.
+                // The imports are required if a consumer of the scripting API is using a Task implementation 
+                // that uses extension methods.
+                var binder = new InContainerBinder(
+                    container: null,
+                    next: new BuckStopsHereBinder(compilation),
+                    imports: compilation.GlobalImports);
```

Comment:
No I meant, in your code, you have `new InContainerBinder(compilation.GlobalNamespace, binder)`, what is the "binder" here?